BuildNotifications/ViewModel/Settings/Setup/ProjectViewModel.cs

```diff
@@ -0,0 +1,110 @@
+
+            var connections = configuration.Connections.Where(c => c.ConnectionType == ConnectionPluginType.SourceControl).ToList();
+            var connection = connections.FirstOrDefault(c => c.Name == model.SourceControlConnectionName);
+            SourceControlConnection = new ConnectionOptionViewModel(StringLocalizer.SourceControlConnectionNames, connections, connection);
+
+            connections = configuration.Connections.Where(c => c.ConnectionType == ConnectionPluginType.Build).ToList();
+            connection = connections.FirstOrDefault(c => c.Name == model.BuildConnectionName.FirstOrDefault());
+            BuildConnection = new ConnectionOptionViewModel(StringLocalizer.BuildConnectionNames, connections, connection);
+
+            foreach (var option in Options)
+            {
+                option.ValueChanged += Option_ValueChanged;
+            }
+        }
+
+        public StringCollectionOptionViewModel BranchBlacklist { get; }
+        public StringCollectionOptionViewModel BranchWhitelist { get; }
+        public ConnectionOptionViewModel BuildConnection { get; }
+        public StringCollectionOptionViewModel BuildDefinitionBlacklist { get; }
+        public StringCollectionOptionViewModel BuildDefinitionWhitelist { get; }
+        public TextOptionViewModel DefaultCompareBranch { get; }
+        public override string DisplayNameTextId => Model.ProjectName;
+        public BooleanOptionViewModel HideCompletedPullRequests { get; }
+        public override IconType Icon => IconType.None;
+        public BooleanOptionViewModel IsEnabled { get; }
+        public IProjectConfiguration Model { get; }
+        public TextOptionViewModel Name { get; }
+
+        public IEnumerable<OptionViewModelBase> Options
+        {
+            get
+            {
+                yield return IsEnabled;
+                yield return Name;
+                yield return BuildConnection;
+                yield return SourceControlConnection;
+                yield return BranchBlacklist;
+                yield return BranchWhitelist;
+                yield return BuildDefinitionBlacklist;
+                yield return BuildDefinitionWhitelist;
+                yield return DefaultCompareBranch;
+                yield return HideCompletedPullRequests;
+                yield return PullRequestDisplayMode;
+            }
+        }
+
+        public PullRequestDisplayModeOptionViewModel PullRequestDisplayMode { get; }
+
+        public ICommand SaveCommand { get; }
+        public ConnectionOptionViewModel SourceControlConnection { get; }
+        public virtual event EventHandler<EventArgs>? SaveRequested;
+
+        private void Option_ValueChanged(object? sender, EventArgs e)
+        {
+            Save();
+        }
+
+        private void RaiseSaveRequested()
+        {
+            SaveRequested?.Invoke(this, EventArgs.Empty);
+        }
+
+        private void Save()
```

Comment:
I don't see how I can serialize something without setting the values to serialize first?
This is standard code that maps view model to model (and vice versa in the constructor)

When a developer adds a new option to the view (i.e. they added it to this ViewModel) they should expect to have a place to read the value from the model and save the value back to the model?

Reply:
Well yeah, I guess that's the difference to the way I think is more efficient to program and in the end the approach ReflectSettings uses.
My though process is as follows:
Let's say you have a property in your ViewModel like this:

`public string SomeProperty {
   get => Model.SomeProperty;
   set => Model.SomeProperty = value;
}`

With no need to explicitly save values, because each value is directly written to the underlying model.
Now, with our Option approach, I could imagine something similar to this:

public TextOptionViewModel SomePropertyOption { get; } = TextOptionViewModel.FromBinding(Model, m => m.SomeProperty);

And the TextOptionViewModel takes care of setting the SomeProperty value. Which makes sense, so the TextOptionViewModel can take care of all the "is my input value correct?" "Does it exceed min/max?" "AvailableValues etc.".


If you WANT an explicit save functionality, I'd suggest to shadow copy the model. So something like:

var copy = (Model) model.MemberwiseClone();
// modify copy
// persist:
JsonConvert.Populate(JsonConvert.Deserialize(copy), model);

This way you bypass all the clutter code that you have to do.
This is what I had in mind when I said "[have this] done implicitly via serialization"

I guess I just personally hate the fact, that I have to write more than 1 line in my ViewModel for something so trivial.
Or more than 1 property.

Maybe I'm wrong and people generally prefer it that way, I just can't seem to find a compelling argument to convince me that explicitly writing the "How the value is loaded, how it is given outside and how it's persisted" is better than just "do it the same way like all the other values. You know how".
In modern Web Dev it is done all the time. 
Anyway, I leave this decision to you. I don't have a strong grudge against doing it the way you wrote. So either way is fine by me.

Reply:
Ah now I see. Well yes that would be nice. However I'd say that this could/should be done on its own. (I just want this PR to be done :D)

Reply:
I see you created #164. I agree, let's get this done already. This is not a reason to block the PR.